libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFRootViewManager.m

```diff
@@ -104,99 +104,72 @@ - (void)enumerateDelegates:(void (^)(id<SFRootViewManagerDelegate> delegate))blo
 // Whatever the implementation, it promises to be complex.  As of iOS 6.1, Apple simply does not make
 // the presentation of an "uber" view easy to implement on the edges.
 //
-
-- (void)pushViewController:(UIViewController *)viewController
-{
-    __weak typeof(self) weakSelf = self;
-    void (^pushControllerBlock)(void) = ^{
-        __strong typeof(weakSelf) strongSelf = weakSelf;
+- (void)pushViewController:(UIViewController *) viewController {
+    
+    if (!viewController)
+        return;
+  
+    __weak typeof (self) weakSelf = self;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        
+        __strong typeof (weakSelf) strongSelf = weakSelf;
         UIViewController *currentViewController = strongSelf.mainWindow.rootViewController;
-        while (currentViewController.presentedViewController != nil && !currentViewController.presentedViewController.isBeingDismissed) {
-            if([currentViewController.presentedViewController isKindOfClass:[UIAlertController class]]) {
-                strongSelf->_modalViewController = (UIAlertController *)currentViewController.presentedViewController;
-                [currentViewController.presentedViewController dismissViewControllerAnimated:NO completion:nil];
+        while (currentViewController.presentedViewController != nil) {
+            //stop if we find that an alert has been presented
+            if ([strongSelf alertIsPresented:currentViewController]) {
+                [strongSelf saveAlert:currentViewController.presentedViewController];
                 break;
             }
             currentViewController = currentViewController.presentedViewController;
         }
         
-        if (currentViewController != nil) {
-            if (currentViewController != viewController
-                && viewController.presentedViewController != currentViewController
-                ) {
-                [strongSelf log:SFLogLevelDebug format:@"pushViewController: Presenting view controller (%@).", viewController];
-                
-                [strongSelf enumerateDelegates:^(id<SFRootViewManagerDelegate> delegate) {
-                    if ([delegate respondsToSelector:@selector(rootViewManager:willPushViewControler:)]) {
-                        [delegate rootViewManager:strongSelf willPushViewControler:viewController];
-                    }
-                }];
-                
-                if([currentViewController isKindOfClass:[UINavigationController class]]) {
```

Comment:
@khawkins I remember that patch. I don't think that it is required. I haven't seen any instance of failures in all the tests I've done so far. I wasn't too confident of the original patch anyway. @brandonpage is helping to test the overall functionality. Lets merge only after the tests.  I will keep you posted.